crates/sui-types/src/inner_temporary_store.rs

```diff
@@ -11,14 +11,16 @@ use crate::{
 use move_binary_format::CompiledModule;
 use move_bytecode_utils::module_cache::GetModule;
 use move_core_types::language_storage::ModuleId;
+use serde::Deserialize;
+use serde::Serialize;
 use std::collections::BTreeMap;
 use std::sync::Arc;
 
 pub type WrittenObjects = BTreeMap<ObjectID, Object>;
 pub type ObjectMap = BTreeMap<ObjectID, Object>;
 pub type TxCoins = (ObjectMap, WrittenObjects);
 
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
```

Comment:
Why do we need this?

Reply:
The sandbox state captures the execution state as at when the sandbox was created. This is useful for debugging.
Indeed it is not needed when replaying it as we only care can just check if effects match so we can remove it.